### PR TITLE
fix(library): stop re-enrichment from silently re-enabling monitoring

### PR DESCRIPTION
**Pull request by arsfeld**
Fixes the self-enabling monitoring reported in https://github.com/getmydia/mydia/issues/653#issuecomment-5501654427: an operator disabled monitoring on a movie, and 22 days later it read Monitored again with nothing in its history recording the change.

This is independent of the re-scan trashing in the body of #653, which is a separate defect with its own fix.

## Cause

`MetadataEnricher.build_media_item_attrs/3` built one attrs map and handed it to two callers with opposite needs: the create path, and the update path for an item already in the library. The map hardcoded `monitored: true`, and `MediaItem.changeset/2` casts `:monitored`, so every re-enrichment overwrote the operator's choice. The only thing in the way was `recently_enriched?/1`, whose window is one hour, so any scan promoting a file for an item last touched longer ago than that rewrote the flag.

It went unseen because `extract_meaningful_changes/2` recorded only title, original_title and year. The write left a `media_item.updated` event whose sole mark was `reason: "Metadata enriched"`. The history was never empty; it just never mentioned monitoring.

The same defect had already been fixed one level down: `Media.upsert_episodes_from_season/3` takes `monitor_new?` through `Keyword.fetch!` with the comment *"A default here is what let the scanner and enricher drift into monitoring everything regardless of the show's intent"*, and its update branch omits `monitored` deliberately. The episode level was hardened; the media item level was missed.

## What changed

**Split the enricher's attrs builder** (`a99ce7c`). The create path keeps `monitored: true`; the update path gets a builder returning only what the metadata provider owns. The update attrs now have no `:monitored` key, and by construction no `:monitor_new_seasons`, `:quality_profile_id` or `:library_path_id` either, so the next field added to the builder cannot leak the same way.

**Record monitoring changes in history** (`828e3cf`, `dbb8794`). `:monitored` and `:monitor_new_seasons` are added to `extract_meaningful_changes/2`, and both presentation layers that render an event's changes were widened to match. Without the second commit the change reached the database but appeared nowhere a person looks: `Presentation.changes_summary/1` and `ActivityLive.format_change_details/1` each hardcoded their own three-field list.

**Restore the already-flipped items** (`67a5769`). `Mydia.Jobs.MonitoringRepair`, a one-shot Oban worker on the `HdrBackfill` pattern, enqueued once at boot. The bug destroyed the setting but not the record of it: the toggles write `reason: "Monitoring enabled"` / `"Monitoring disabled"` into event metadata, retained 90 days by default. An item is restored when two facts hold, that it is monitored now and its most recent recorded monitoring decision was a disable.

It is idempotent with no stamp column and therefore no migration: the restoration emits its own `monitoring_changed` event, which becomes the item's latest decision, and the item is no longer monitored, so a second pass cannot select it. A manual re-enable afterwards is likewise a later decision, so it never undoes the operator.

`Event.metadata` is a map serialized into a TEXT column on both SQLite and PostgreSQL, so there is no portable JSON operator for it. The query narrows by type and resource in chunks of 200 ids; the reason match runs in Elixir.

**Closed a second silent path** (`544a30d`). Found by review, not by the original design. `Media.update_media_items_batch/2`, reached from the Batch Edit modal, set `monitored` through a bare `Repo.update_all` with no event at all, the same silent shape as the enricher bug. That falsified the premise the repair worker rests on: an operator who disabled an item with the single toggle and later re-enabled it via Batch Edit would leave exactly the state the worker treats as evidence of the bug, and the repair would have undone their deliberate re-enable. `update_media_items_batch/3` now records one `monitoring_changed` event per affected item when, and only when, the batch sets `monitored`.

## Limits worth knowing

Items whose disable predates the 90 day event retention window carry no record and are left alone. They cannot be recovered, and operators should re-check anything they unmonitored longer ago than that.

`events.inserted_at` is second-precision and `events.id` is a random UUID, so two decisions recorded in the same second order arbitrarily. Not a real production scenario, and the worst case is one item left monitored, fixable from the UI.

Unlike `HdrBackfill`, this worker has no stamp column, so its per-boot cost does not fall to zero once the backlog drains. It re-derives the candidate set each boot. Cheap on a typical self-hosted library; documented honestly in the boot comment rather than borrowing `HdrBackfill`'s "harmless once drained" claim, which does not hold here.

## Testing

Full suite green on this branch: 13 doctests, 10103 tests, 0 failures, 44 skipped.

New coverage: an unmonitored item survives re-enrichment unmonitored (movie and show), a new item is still created monitored, the update still writes the fields it owns; monitored changes appear in event change details and render in the Activity Feed; the repair worker's selection rule across seven cases including idempotency and a manual re-enable; batch edit emits one decision per item when it sets monitored and none otherwise.

## Follow-up, not in this PR

`Media.set_monitor_new_seasons/2` writes via `Repo.update_all` and calls `Events.media_item_updated/5` with no changes argument, so `changes["monitor_new_seasons"]` never populates from the real call path. The rendering added here for that field is currently unreachable by user action. Harmless, but worth closing separately.

## Summary by CodeRabbit

- **Bug Fixes**
  - Re-enriching media no longer unintentionally re-enables monitoring settings.
  - Existing items with incorrect monitoring states are automatically corrected without overriding newer manual decisions.
  - Monitoring decisions are now consistently recorded in activity history.
  - Newly added media continues to receive the appropriate default monitoring setting.

- **Improvements**
  - Batch monitoring changes generate clear per-item activity events.
  - Activity details now show readable monitoring and new-season monitoring changes.

### CODE REVIEW

File: lib/mydia/jobs/monitoring_repair.ex [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  def perform(%Oban.Job{}) do
+    batch_size = Application.get_env(:mydia, :monitoring_repair_batch_size, @default_batch_size)
+
+    case pending_ids(batch_size) do
+      [] ->
+        Logger.info("Monitoring repair complete, no items pending")
+        :ok
+
+      ids ->
+        Enum.each(ids, &repair_one/1)
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia/jobs/monitoring_repair.ex:193]: Valid, and worth fixing precisely because it is the same class of bug this PR exists to close. Fix in progress: `repair_one/1` will re-evaluate the item's latest decision inside a transaction with the write, skip silently when a newer decision has appeared, and emit the `monitoring_changed` event only when the guarded update actually happened.

Worth noting the recheck is only reliable because of the last commit on this branch: `update_media_items_batch/3` previously set `monitored` through a bare `Repo.update_all` with no event, so a concurrent enable through Batch Edit would have left no newer decision for a recheck to find.

File: lib/mydia/events/presentation.ex [hunk 2]
```diff
@@ -569,7 +569,7 @@ defmodule Mydia.Events.Presentation do
 
     simple_fields =
       changes
-      |> Map.take(["title", "original_title", "year"])
+      |> Map.take(["title", "original_title", "year", "monitored", "monitor_new_seasons"])
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia/events/presentation.ex:572]: Agreed, the two renderers disagreeing is a real inconsistency. Fixing so the short summary and the expanded Activity Feed use the same labels.

File: test/mydia/jobs/monitoring_repair_test.exs [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      # This is exactly what pending_ids/1 selects on.
+      assert MonitoringRepair.pending_ids(10) == [item.id]
+
+      # Simulates the race directly: an operator re-enables monitoring,
+      # recording a newer decision, between selection and repair_one/1's
+      # write. No sleeps or concurrent processes needed, since repair_one/1
+      # re-reads the decision itself immediately before writing.
+      record_update(item, "Monitoring enabled", 120)
+
+      assert :ok = MonitoringRepair.repair_one(item.id)
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply test/mydia/jobs/monitoring_repair_test.exs:164]: Correct, and a real gap in the previous fix rather than a theoretical one. The update moved inside the transaction but the event write stayed outside it, so a re-enable landing in that window leaves the repair's disable as the newest decision on a monitored item, which `pending_ids/1` then reselects. Moving the event into the transaction. This works because `Events.create_event_async/1` checks `Repo.in_transaction?()` and writes synchronously on the same connection, so it commits and rolls back with the update.

File: lib/mydia/events/presentation.ex [hunk 4]
```diff
@@ -579,7 +580,7 @@ defmodule Mydia.Events.Presentation do
   end
 
   defp summarize_fields(fields) do
-    names = fields |> Enum.take(3) |> Enum.map(&field_name/1)
+    names = fields |> Enum.take(3) |> Enum.map(&(&1 |> field_name() |> field_label()))
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia/events/presentation.ex:583]: Fair catch on the fallback I introduced. `field_label/1` delegating to `Phoenix.Naming.humanize/1` widened the input contract to something the persisted-event path cannot guarantee, and a rendering helper should not be able to take down the whole Activity Feed over one malformed row. Adding a guard for non-binary, non-atom values, with tests for nil and a numeric field.

**Comment by arsfeld** [comment]
@coderabbitai review

Both findings from the last round are addressed in f9b8549, which landed after your review of 0a18494:

- The repair's decision event now writes inside `repair_one_tx/1`'s transaction, so the guarded update and its `monitoring_changed` event commit or roll back together. `Events.create_event_async/1` writes synchronously when `Repo.in_transaction?/0` is true, so there is no longer a window where the update lands without its event.
- `Presentation.field_label/1` now has explicit binary and non-nil-atom clauses and falls back to "Unknown field" for anything else, so a malformed persisted event cannot reach `Phoenix.Naming.humanize/1` and take down event detail or the Activity Feed. `field_name/1`'s map clause is guarded with `is_binary` too.

The merge-risk summary still reads "up to 0a184", so it predates both fixes.
